cpp/src/arrow/compute/exec/asof_join_benchmark.cc

```diff
@@ -112,8 +115,8 @@ AsofJoinNodeOptions GetRepeatedOptions(size_t repeat, FieldRef on_key,
 
 static void AsOfJoinOverhead(benchmark::State& state) {
   int64_t tolerance = 0;
-  AsofJoinNodeOptions options =
-      GetRepeatedOptions(int(state.range(4)), kTimeCol, {kKeyCol}, tolerance);
+  auto options = std::make_shared<AsofJoinNodeOptions>(
+      GetRepeatedOptions(int(state.range(4) + 1), kTimeCol, {kKeyCol}, tolerance));
```

Comment:
@rtpsw can you verify this change?  `state.range(4)` is the # of right tables but it seems we need a `Keys` element for each of the right tables as well as 1 for the left table.

Reply:
Sorry, I somehow missed this. The fix seems right to me.